Account/SyncAuthState.swift

```diff
@@ -21,6 +21,7 @@ public struct SyncAuthStateCache {
 public protocol SyncAuthState {
     func invalidate()
     func token(_ now: Timestamp, canBeExpired: Bool) -> Deferred<Maybe<(token: TokenServerToken, forKey: Data)>>
+    var deviceRegistration: FxADeviceRegistration? { get }
```

Comment:
Can we narrow this to be `deviceID` instead of `deviceRegistration`? We don't use any other aspect of the `FxADeviceRegistration`.

Reply:
Is there anything else we might need here in `FxADeviceRegistration` in the future?

Reply:
There are two other fields: the version of the device registration, and the last registration timestamp. Neither of those should be relevant to Sync; they're just there to control when we re-register.

Reply:
:+1: